application-task-default/src/main/resources/templates/notification/com.xwiki.task.internal.notifications.TaskChangedEvent.vm

```diff
@@ -0,0 +1,94 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+
+#template('notification/macros.vm')
+
+#set ($mainIcon = 'list')
+#set ($smallIcon = '')
+
+#macro (displayEventDetails $event)
+  #set ($document = $xwiki.getDocument($event.document))
+  #set ($customEventInfo = $jsontool.fromString($event.getBody()) )
+  #if ("$!customEventInfo" == '')
+    ## Do nothing if the json is either malformed, or the body contains something else (like if the event is part of
+    ## another application, and it was grouped with the notifications from TaskManager).
```

Comment:
Should investigate why this happens, how to exclude notifications from other sources (like the default page created event) from being grouped with `TaskChangedEvent`. This leads to warnings in the logs when rendering the notifications:
```
2024-11-11 11:38:01,288 [notifications job group daemon thread - org.xwiki.rendering.async.internal.AsyncRendererJob@7c4130a5([notifications, display, 5ALERT_17xwiki:XWiki.Admin_10_null_-457216359, locale, en, request.wiki, xwiki, user, xwiki:XWiki.Admin, 68])] INFO  o.x.v.t.JSONTool               - Failed to parse JSON [activitystream.event.create]: JsonParseException: Unrecognized token 'activitystream': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
```

Reply:
Can you provide some steps to reproduce this use case, or some examples of what happens in terms of UI?
If I understand correctly, you mean that if a task was updated along with other page changes, these events are grouped. But what template is used for displaying them?

Reply:
1. Make sure you are subscribed to both TaskManager and Page created/updated events (user administration page)
2. Watch the entire wiki for notifications
3. Create/Update a task
4. Check the notifications panel.

For this problem, you should only see the TaskManager events, with no sign of the page updated/created event. In the console log of the server there should be the message in the first comment in this thread. The UI has no major visual breakages. The broken things are the missing event and the logs.

While taking the screenshots I found out that when listening to both TaskManager and Pages events, one event gets grouped 'alone' with the Pages event, while the others are grouped together (see first picture, it's always the last sent event, namely `status`.

The TaskManager template is used to display all grouped events, which is the cause for the failed JSON parse in the log, which is why the page event doesn't appear at all.

When a task is updated, its page is also updated, which leads to the chain reaction of producing an XObjectUpdated and a DocumentUpdated event, which both appear as notifications (TaskManager and Pages apps).

What it looks like:
![image](https://github.com/user-attachments/assets/550939e9-c654-41fe-b25a-882390bc7879)

With debug statement (page event for the same page grouped with task notification):
![image](https://github.com/user-attachments/assets/1296a348-01cd-4123-8c37-ed8c9fbcfc32)

Reply:
To add filter for the event type in `details` macro (move it from `displayEventDetails`)